### ISSUE TITLE
Possible to make spring TaskExecutor extend ExecutorService or provide similar functions on TaskExecutor? [SPR-8794]

### ISSUE THREAD

**Issue by spring-issuemaster** [spring-projects/spring-framework#13436]
**[pas filip](https://jira.spring.io/secure/ViewProfile.jspa?name=pasfilip)** opened **[SPR-8794](https://jira.spring.io/browse/SPR-8794?redirect=false)** and commented

Hi,
I was wondering why spring's task executor didn't also extend from java.util.ExecutorService and if it could be added?
The only reason I could think of is that it has grown as such historically or perhaps that there is a backwards compatible issue somewhere or perhaps it was
chosen because the shutdown methods didn't want to be exposed?
Not sure why and maybe this has already been discussed internally or on mailing lists etc although I didn't find anything related to this topic after a brief search.
I apologize in advance if it's an outdated discussion.
It looks like it shouldn't be too difficult to implement this in most TaskExecutor implementations and it would be a useful extension if you want to inject an ExecutorService in a managed bean making use of the invokeAll, invokeAny methods regardless of the underlying infrastructure being managed or unmanaged.
The motivation I have for wanting an ExecutorService is that it isn't DRY code to achieve the semantics provided by invokeAll or invokeAny using the Async annotation or the TaskExecutor.
The drill is to create the Callable or Runnable and submit them or invoke the async methods, collect the futures and iterate over them to retrieve the results.(could use static methods to be DRY but still not ideal)
It would be nice to be able to inject the ExecutorService into your bean and just invoke invokeAll for the situations where a bunch of jobs need to run and all return while using the same pool underneath for this as used for the Async annotation.
Nothing actually prevents me from doing this it just means I can't use the nice task namespace to keep the xml configuration short.
If the TaskExecutor would extend ExecutorService I could use the task namespace and inject a ExecutorService into any managed by without needing to update the xml config. (All I would need to add is a Qualifier or custom Annotation for discriminating with the scheduler pool should there be one)
It would also mean I could use the WorkManager implementation without having to write an adapter to provide the invokeAll and invokeAny semantics.
Alternatively I could retrieve the ExecutorService that some of spring's TaskExecutor implementations contain and expose it as a bean but it would mean I couldn't switch easily to using alternate implementations of TaskExecutor that don't contain a wrapped ExecutorService.

In summary I'd say it would be a nice addition which provides a useful service while keeping configuration changes to a minimum and providing more DRY code.

---

1 votes, 2 watchers

**Comment by spring-issuemaster** [comment]
Bulk closing outdated, unresolved issues. Please, reopen if still relevant.
